### PR TITLE
Close File Installation Log

### PR DESCRIPTION
**Pull request by datamel**
Turns out the file installation log is not closed properly. 

This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (strange one to test, attempted to prove closure by reloading and checking that a new file is opened, which is the desired behaviour.).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/task_remote_mgr.py
Comment (oliver-sanders) [inline cylc/flow/task_remote_mgr.py:526]: I think _file_install_callback gets called once per remote-init so if we are remote-init'ing multiple platforms the logs will get closed multiple times?

Sadly I think because of that this is hard to do.

E.G. if we have three platforms the order of events would look something like this:

1. open logs
2. remote-init x
3. remote-init y
4. remote-init z
5. close logs

Where we need to wait for all remote-inits to succeed before closing the log.

Best option I can think of is to close the file install logs on shutdown and reload. On reload we would also need to open the next file-install log. Potential problem with that is if a remote-init is in progress when the reload is started it will try to log to a closed handler which could cause issues.

Comment (datamel) [reply cylc/flow/task_remote_mgr.py:526]: Yes, good spot. I take it we shouldn't have a file installation log per platform?

Comment (oliver-sanders) [reply cylc/flow/task_remote_mgr.py:526]: We could stick them into one file or break them apart into separate ones, both would be valid.

If we were to break them apart into separate files then we no longer need to use `logging` (because we can just `with open(<file>)`, there's only one entry to write per file).

Comment (oliver-sanders) [reply cylc/flow/task_remote_mgr.py:526]: (one to raise at the team meet tomorrow)

Comment (hjoliver) [reply cylc/flow/task_remote_mgr.py:526]: (one per install target seems a good idea to me)

Comment (oliver-sanders) [reply cylc/flow/task_remote_mgr.py:526]: Added this in https://github.com/cylc/cylc-flow/issues/4802#issuecomment-1091424254

**Comment by datamel** [comment]
This log becomes redundant in https://github.com/cylc/cylc-flow/pull/4836. Closing it for now as quick fix for rc3.

**Review by hjoliver — APPROVED** [review]

**Comment by hjoliver** [comment]
> This log becomes redundant in https://github.com/cylc/cylc-flow/pull/4836. Closing it for now as quick fix for rc3.

OK cool. I can't see any way for this small change to cause trouble; no need for a second review.

**Pull request merged**
